src/pyfaf/opsys/centos.py

```diff
@@ -192,11 +192,22 @@ def get_releases(self):
     def get_components(self, release):
         urls = [repo.replace("$releasever", release)
                 for repo in [self.base_repo_url, self.updates_repo_url]]
-
-        yum = Yum(self.name, *urls)
-        components = list(set(pkg["name"]
-                              for pkg in yum.list_packages(["src"])))
-        return components
+        components = []
+        if "dnf" in repo_types:
+            from pyfaf.repos.dnf import Dnf
+            dnf = Dnf(self.name, *urls)
+            components.extend(list(set(pkg["name"]
+                                       for pkg in dnf.list_packages(["src"]))))
+            return components
```

Comment:
pylint is complaining about else after return. Would be better to move the return after conditions.

Reply:
You are right, thanks, I will fix.